common/src/chain/pow.rs

```diff
@@ -0,0 +1,112 @@
+use crate::chain::config::ChainType;
+use crate::chain::ChainConfig;
+use crate::Uint256;
+use std::time::Duration;
+
+/// Chain Parameters for Proof of Work, as found in
+/// https://github.com/bitcoin/bitcoin/blob/eca694a4e78d54ce4e29b388b3e81b06e55c2293/src/chainparams.cpp
```

Comment:
Similar comment like below. To make it render nicely in the docs, maybe have a description and the link in the detailed section separately, roughly like that:
```
/// Chain Parameters for Proof of Work.
///
/// As found in Bitcoin [chainparams.cpp](https://github.com/bitcoin/bitcoin/blob/eca694a4e78d54ce4e29b388b3e81b06e55c2293/src/chainparams.cpp)
```